src/index.ts

```diff
@@ -1,45 +1,186 @@
+};
+
 module _ {
-	export function fetch() {
-		return {};
-	}
+  export function fetch(
+    resource: FetchResource,
+    options?: FetchOptions
+  ): Promise<Response> {
+    return fetch(resource, options);
+  }
+
+  export function isNull(value: unknown): boolean {
+    return value === null;
+  }
+
+  export function isNil(value: unknown): boolean {
+    return value === null || value === undefined;
+  }
+
+  export function isNumber(value: unknown): boolean {
+    return typeof value === "number";
+  }
+
+  export function isFunction(value: unknown): boolean {
+    return typeof value === "function";
+  }
 
-	export function isNull() {}
+  export function shuffle<T>(array: T[]): T[] {
+    const length = array === null ? 0 : array.length;
+    if (!length) return [];
+    let index = -1;
+    const lastIndex = length - 1;
+    const result = [...array];
+    while (++index < length) {
+      const rand = index + Math.floor(Math.random() * (lastIndex - index + 1));
+      const value = result[rand];
+      result[rand] = result[index];
+      result[index] = value;
+    }
+    return result;
+  }
 
-	export function isNil() {}
+  export function pick<
+    T extends Record<string, unknown>,
+    K extends (keyof T)[]
+  >(object: T, paths: K): PickResult<T, K> {
+    const result = paths.reduce(
+      (prev, cur) => ({ ...prev, [cur]: object[cur] }),
+      {} as T
+    );
+    return result;
+  }
 
-	export function isNumber() {}
+  export function omit<
+    T extends Record<string, unknown>,
+    K extends (keyof T)[]
+  >(object: T, paths: K): OmitResult<T, K> {
+    const result = { ...object };
+    paths.forEach((key) => {
+      delete result[key];
+    });
+    return result;
+  }
 
-	export function isFunction() {}
+  export function memoize<T, R>(
+    func: Function<T, R>,
+    resolver?: Function<T, string>
+  ): Function<T, R> {
+    const memoized = function (args: any) {
+      const key = resolver ? resolver(args) : args[0];
+      const cache = new Map();
 
-	export function shuffle() {}
+      if (cache.has(key)) {
+        return cache.get(key);
+      }
 
-	export function pick() {}
+      const result = func(args);
+      cache.set(key, result);
+      return result;
+    };
+    return memoized;
+  }
 
-	export function omit() {}
+  export function debounce<T, R>(
+    func: Function<T, R>,
+    wait: number
+  ): Function<T, void> {
+    let timerId: NodeJS.Timeout | null;
 
-	export function memoize() {}
+    return (...args) => {
+      if (timerId) clearTimeout(timerId);
+      timerId = setTimeout(func, wait, ...args);
+    };
+  }
 
-	export function debounce() {}
+  export function throttle<T, R>(
+    func: Function<T, R>,
+    wait: number
+  ): Function<T, void> {
+    let timerId: NodeJS.Timeout | null;
 
-	export function throttle() {}
+    return (...args) => {
+      if (timerId) return;
+      timerId = setTimeout(() => {
+        func(...args);
+        timerId = null;
+      }, wait);
+    };
```

Comment:
c: 타입과 관련된 사항은 아니지만 마지막으로 호출된 throttled 함수의 인자로 실행하기 위해서는 클로저를 이용해 마지막 인자를 참조해야합니다.

```suggestion
    let lastArgs: {
      current: Parameters<T> | [];
    } = {
      current: [],
    };
    
    return (...args) => {
      lastArgs.current = args;
      if (timerId) return;
      timerId = setTimeout(() => {
        func(...lastArgs.current);
        timerId = null;
      }, wait);
    };
```

Reply:
고마워요 우디 다시 한번 확인해볼게요~ 👍